### PR TITLE
CFD-329 : formatting bug on vocabulary lists

### PR DESCRIPTION
**Pull request by nader77**
Fix formatting.
edit_discussion___capacity4more

### CODE REVIEW

**Comment by nader77** [comment]
@HelenaEksler Could you review this please ?

**Comment by nader77** [comment]
Do we need to have the check_plain ?

**Comment by HelenaEksler** [comment]
@nader77 I think we need some check for cases when user might set the term name with some extra symbols, html tags or smth.

**Comment by nader77** [comment]
@amitaibu @HelenaEksler Can i user ``strip_tags`` ?

**Comment by nader77** [comment]
@amitaibu ^^

File: capacity4more/modules/c4m/vocab/c4m_vocab_widget/c4m_vocab_widget.module [hunk 1]
```diff
@@ -166,7 +166,7 @@ function c4m_vocab_widget_prepare_taxonomy_tree($terms) {
     // If the term is array, this vocab is of type "node".
     if(is_array($term)) {
       foreach($term as $id => $label) {
-        $options[$id]['label'] = check_plain($label);
+        $options[$id]['label'] = $label;
```
Comment (amitaibu) [inline capacity4more/modules/c4m/vocab/c4m_vocab_widget/c4m_vocab_widget.module]: ``filter_xss()``

**Comment by nader77** [comment]
@amitaibu @HelenaEksler  Added a special chars decode function on the check_plain, This will make it print in the proper way, This is common on select lists. 
edit_discussion___capacity4more

**Comment by nader77** [comment]
Update: replaced ``check_plain`` with ``filter_xss`` because the latter will remove the XSS chars instead of replacing them with special chars and in that case the ``htmlspecialchars_decode`` will only decode the non XSS chars Like ampersand(&).
e.g:
perl_-e__print___img_src_java_0script_alert___xss__________out___capacity4more
edit_discussion___capacity4more

OR:
worldwide_img_src_javascript_alert__xss______capacity4more
edit_discussion___capacity4more

**Comment by nader77** [comment]
@amitaibu @HelenaEksler PR is ready for code review.

**Comment by HelenaEksler** [comment]
@nader77 Looks ok.

**Pull request merged**